hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -808,6 +808,7 @@ public void fixTasksForFailedHost(List<HostState> hosts, String failedHost) {
 
     private List<JobTask> findAllTasksAssignedToHost(String failedHostUUID) {
         List<JobTask> rv = new ArrayList<>();
+        // Why?
```

Comment:
I think that it is prudent to hold the job lock while failing hosts -- which is the only time this method is called. We don't want to, e.g., kick tasks before all the bad hosts have been replaced.

However, this read-only method does not appear to be useful place to get the lock. I would suggest SpawnBalancer's `attemptFixTaskForFailedHost` as a method that more clearly needs the lock.